packages/wrangler/src/deployment-bundle/bundle.ts

```diff
@@ -278,6 +287,42 @@ export async function bundleWorker(
 		inject.push(path.resolve(getBasePath(), "templates/modules-watch-stub.js"));
 	}
 
+	// esbuild's `alias` option is applied after each plugin's onResolve hook,
+	// whereas we would like these user-defined aliases to take precedence over
+	// the unenv polyfill aliases, so we reimplement the aliasing as a plugin
+	// to be applied before that plugin (earlier in the array of plugins)
+	const aliasPlugin: esbuild.Plugin = {
+		name: "alias",
+		setup(build) {
+			if (!alias) {
+				return;
+			}
+
+			// filter the hook calls to only those that match the alias keys
+			// this should avoid slowing down builds which don't use aliasing
+			const filter = new RegExp(
+				Object.entries(alias)
+					.map(([key, _value]) => escapeRegex(key))
+					.join("|")
+			);
```

Comment:
If we don't need the values then can we just use:

```suggestion
				Object.keys(alias)
					.map((key) => escapeRegex(key))
					.join("|")
			);
```